Xamarin.Forms.Build.Tasks/DebugXamlCTask.cs

```diff
@@ -113,8 +113,16 @@ public override bool Execute(out IList<Exception> thrownExceptions)
 						var br2 = Instruction.Create(OpCodes.Ldarg_0);
 						var ret = Instruction.Create(OpCodes.Ret);
 						il.Emit(OpCodes.Ldarg_0);
-						var baseCtor = module.ImportReference(typeDef.BaseType.Resolve().GetConstructors().First(c => c.HasParameters == false));
-						baseCtor = module.ImportReference(baseCtor.ResolveGenericParameters(typeDef.BaseType, module));
+						MethodReference baseCtor;
+						if (typeDef.BaseType.Resolve().GetConstructors().FirstOrDefault(c => c.HasParameters && c.Parameters.Count == 1 && c.Parameters[0].Name == "useCompiledXaml") != null) {
+							baseCtor = module.ImportReference(typeDef.BaseType.Resolve().GetConstructors().First(c => c.HasParameters && c.Parameters.Count == 1 && c.Parameters[0].Name == "useCompiledXaml"));
```

Comment:
Can we simplify this so we don't have to repeat this Linq query for the constructor?

Reply:
not a critical path (this code is only used when compiling Unit Tests), but you're right

Reply:
fixed